### DISCUSSION TITLE
How to send an event to an unrelated machine

### DISCUSSION THREAD

**Discussion by AntonS86** [statelyai/xstate#1773] [Q&A] [answered]
https://stackoverflow.com/questions/65430799/how-to-send-an-event-to-an-unrelated-machine-xstate

I cannot send an event to an unrelated machine, there is no way to nest one into another, the level of hierarchies can be very large!

machines are not related!!!

```js
Machine({
   id: "first",
   initial: "idle",
   states: {
      idle: {}
   },
   on: {
      EVENT: {actions: "someActions"}
   }
});

Machine({
   id: "second",
   initial: "idle",
   states: {
      idle: {}
   },
   on: {
      "SOME_EVENT": {
         actions: send((ctx, e) => ({...e, data: ctx.some, type: "EVENT",}),
            {to: "first"})
     }
   }
})
```

**Comment by Andarist** [comment, marked as answer]
You have to pass the address that could be used as the value of `to` down - there is no magical global namespace that you could just reach into. Currently, there is no easy way to obtain the address in the service creator though (you could create an issue about it so we could discuss adding it and track this) so you have to work around this by sending an event and grabbing its `_event.origin`.

**Reply by AntonS86** [reply]
Thank you, we will post a question for new functionality

**Reply by ycaptain** [reply]
Was the issue created with any progress? I also want this feature. But I didn't find it anywhere.
Plz tell me if it exists or let me post an issue for it.

**Reply by davidkpiano** [reply]
@ycaptain Feel free to post an issue 🙏

**Reply by ycaptain** [reply]
I have created another discussion for the new feature, the link is https://github.com/davidkpiano/xstate/discussions/2014
